nutils/log.py

```diff
@@ -487,27 +478,44 @@ def range( title, *args ):
   '''Progress logger identical to built in range'''
 
   items = _range( *args )
-  return _logiter( title, _iter(items), length=len(items), useitem=True )
+  log = _getlog()
+  for item in items:
+    with log.context( '{} {} ({:.0f}%)'.format( title, item, (item+.5) * 100. / len(items) ) ):
+      yield item
```

Comment:
Agreed, if you mean to suggest adding 100 instead of 50. Actually the reason I started touching this code (and remove the `useitem` argument) was to log the item generation inside the context, in which case we should add none at all. See new commit 9ffcde2 where I implement this idea.